src/main/java/org/infinispan/persistence/cassandra/CassandraStore.java

```diff
@@ -0,0 +1,322 @@
+   private PreparedStatement selectStatement;
+   private PreparedStatement containsStatement;
+   private PreparedStatement selectAllStatement;
+   private PreparedStatement deleteStatement;
+   private PreparedStatement sizeStatement;
+   private PreparedStatement clearStatement;
+
+   @Override
+   public void init(InitializationContext initializationContext) {
+      ctx = initializationContext;
+      timeService = ctx.getTimeService();
+      marshaller = ctx.getMarshaller();
+      marshalledEntryFactory = ctx.getMarshalledEntryFactory();
+      byteBufferFactory = ctx.getByteBufferFactory();
+      configuration = ctx.getConfiguration();
+   }
+
+   @Override
+   public void start() {
+      try {
+         PoolingOptions poolingOptions = new PoolingOptions();
+         CassandraStoreConnectionPoolConfiguration poolConfig = configuration.connectionPool();
+         poolingOptions.setPoolTimeoutMillis(poolConfig.poolTimeoutMillis());
+         poolingOptions.setHeartbeatIntervalSeconds(poolConfig.heartbeatIntervalSeconds());
+         poolingOptions.setIdleTimeoutSeconds(poolConfig.idleTimeoutSeconds());
+
+         QueryOptions queryOptions = new QueryOptions();
+         queryOptions.setConsistencyLevel(configuration.consistencyLevel());
+         queryOptions.setSerialConsistencyLevel(configuration.serialCconsistencyLevel());
+
+         Cluster.Builder builder = Cluster.builder();
+         builder.withPoolingOptions(poolingOptions);
+         builder.withQueryOptions(queryOptions);
+
+         ArrayList<InetSocketAddress> servers = new ArrayList<>();
+         for (CassandraStoreServerConfiguration cassandraStoreServerConfiguration : configuration.servers()) {
+            InetAddress host = InetAddress.getByName(cassandraStoreServerConfiguration.host());
+            int port = cassandraStoreServerConfiguration.port();
+            servers.add(new InetSocketAddress(host, port));
+         }
+         builder.addContactPointsWithPorts(servers);
+
+         cluster = builder.build();
+
+         if (configuration.autoCreateKeyspace()) {
+            createKeySpace();
+         }
+         session = cluster.connect(configuration.keyspace());
+         entryTable = configuration.entryTable();
+         writeStatement = session.prepare("INSERT INTO " + entryTable + " (key, value, metadata) VALUES (?, ?, ?) USING TTL ?");
+         deleteStatement = session.prepare("DELETE FROM " + entryTable + " WHERE key=?");
+         selectStatement = session.prepare("SELECT value, metadata FROM " + entryTable + " WHERE key=?");
+         containsStatement = session.prepare("SELECT key FROM " + entryTable + " WHERE key=?");
+         selectAllStatement = session.prepare("SELECT key, value, metadata FROM " + entryTable);
+         sizeStatement = session.prepare("SELECT count(*) FROM " + entryTable);
+         clearStatement = session.prepare("TRUNCATE " + entryTable);
+      } catch (Exception e) {
+         log.errorCommunicating(e);
```

Comment:
Best way to do this is to change the "void errorCommunicating(e)" method signateu to return a CacheConfigurationException and then you can "throw log.errorCommunicating(e)". Same for all others